components/wifi/esp8266-driver/ESP8266/ESP8266.cpp

```diff
@@ -584,12 +587,20 @@ nsapi_error_t ESP8266::send(int id, const void *data, uint32_t amount)
             if (_serial_rts == NC) {
                 while (_parser.process_oob()); // Drain USART receive register
             }
+            _busy_timeout_reached = false;
             _smutex.unlock();
             return NSAPI_ERROR_OK;
         }
+        tr_debug("sending failed %d", i);
         if (_error) {
             _error = false;
         }
+        if (_busy && _busy_timeout_reached) {
+            tr_debug("returning NSAPI_ERROR_WOULD_BLOCK");
+            _busy_timeout_reached = false;
+            _smutex.unlock();
```

Comment:
set_timeout() is missing from here. Doesn't make a difference as the default and SEND-timeout are basically the same...